Class1.cs

```diff
@@ -0,0 +1,203 @@
+            }
+            _x = x;
+            _y = y;
+            if (_x < 0 && _y < 0)
+            {
+                _x = -_x;
+                _y = -_y;
+            }
+            this.simplify();
+        }
+
+        public long GetX()
+        {
+            return _x;
+
+        }
+
+        public void SetX(long x)
+        {
+            _x = x;
+            if (_x < 0 && _y < 0)
+            {
+                _x = -_x;
+                _y = -_y;
+            }
+
+        }
+
+        public long GetY()
+        {
+            return _y;
+        }
+
+        public void SetY(long y)
+        {
+            try
+            {
+                if (y == 0)
+                    throw new DivideByZeroException();
+            }
+            catch (DivideByZeroException)
+            {
+
+                Console.WriteLine("No divide by zero");
+            }
+            _y = y;
+            if (_x < 0 && _y < 0)
+            {
+                _x = -_x;
+                _y = -_y;
+            }
+
+        }
+
+        public void simplify()
+        {
+            long a = _x;
+            long b = _y;
+            while (b != 0)
+            {
+                long k = a;
+                a = b;
+                b = k % b;
+            }
+            _x /= a;
+            _y /= a;
+        }
+
+        public double value()
+        {
+            double v = Convert.ToDouble(_x) / Convert.ToDouble(_y);
+            return v;
+        }
+
+
+        public static Fraction operator +(Fraction d1, Fraction d2)
+        {
+            Fraction res = new Fraction();
+            res.SetX((d1._x * d2._y) + (d1._y * d2._x));
+            res.SetY(d1._y * d2._y);
+            res.simplify();
+            return res;
+        }
+
+        public static Fraction operator -(Fraction d1, Fraction d2)
+        {
+            Fraction res = new Fraction();
+            res.SetX((d1._x * d2._y) - (d1._y * d2._x));
+            res.SetY(d1._y * d2._y);
+            res.simplify();
+            return res;
+        }
+
+        public static Fraction operator *(Fraction d1, Fraction d2)
+        {
+            Fraction res = new Fraction();
+            res.SetX(d1._x * d2._x);
+            res.SetY(d1._y * d2._y);
+            res.simplify();
+            return res;
+        }
+
+        public static Fraction operator /(Fraction d1, Fraction d2)
+        {
+            try
+            {
+                if (d2._x == 0)
+                    throw new DivideByZeroException();
+            }
+            catch (DivideByZeroException)
+            {
+
+                Console.WriteLine("No divide by zero");
+            }
+            Fraction res = new Fraction();
+            res.SetX(d1._x * d2._y);
+            res.SetY(d1._y * d2._x);
+            res.simplify();
+            return res;
+        }
+
+        public static bool operator ==(Fraction d1, Fraction d2)
+        {
+            return (d1.value() == d2.value());
+        }
```

Comment:
Consider comparing them fairly (numerator = numerator when having equal denominators). Conversion to double looses precision